heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java

```diff
@@ -161,6 +161,13 @@ private void updateTopology(final PackingPlans.PackingPlan existingProtoPackingP
       Set<PackingPlan.ContainerPlan> containersToAdd = containerDelta.getContainersToAdd();
       Set<PackingPlan.ContainerPlan> containersAdded =
           scalableScheduler.get().addContainers(containersToAdd);
+      if (containersAdded.size() != containersToAdd.size()) {
+        updatedContainers.addAll(containersAdded);
+        throw new RuntimeException("Scheduler failed to add expected countainer count: added "
```

Comment:
"Scheduler failed to add requested containers. Requested X, added Y. ..."

Reply:
updated